backend/app/services/rag/retrieval_service.py

```
@@ -157,3 +164,167 @@ async def retrieve(
+        # Get retriever CRD
+        retriever = retriever_kinds_service.get_retriever(
+            db=db,
+            user_id=user_id,
+            name=retriever_name,
+            namespace=retriever_namespace,
+        )
+
+        if not retriever:
+            raise ValueError(
+                f"Retriever {retriever_name} (namespace: {retriever_namespace}) not found"
+            )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)